runtime/ptrace_check.cc

```diff
@@ -114,7 +115,65 @@ static bool EnableTracingManually(const Settings& vm_settings) {
   return true;
 }
 
+static bool IsTracingCheckUnnecessaryOnVersion() {
+#if !OS_IOS || TARGET_OS_SIMULATOR
+  return true;
+#endif
+
+  // Check for arm64e.
+  cpu_type_t cputype = 0;
+  size_t cputype_size = sizeof(cpu_type_t);
+  if (::sysctlbyname("hw.cputype", &cputype, &cputype_size, nullptr, 0) < 0) {
+    FML_LOG(ERROR) << "Could not execute sysctl() to get CPU type: "
+                   << strerror(errno);
+  }
+
+  cpu_subtype_t cpusubtype = 0;
+  if (::sysctlbyname("hw.cpusubtype", &cpusubtype, &cputype_size, nullptr, 0) <
+      0) {
+    FML_LOG(ERROR) << "Could not execute sysctl() to get CPU subtype: "
+                   << strerror(errno);
+  }
+
+  // Tracing is necessary unless the device is arm64e (A12 chip or higher).
+  if (cputype != CPU_TYPE_ARM64 || cpusubtype != CPU_SUBTYPE_ARM64E) {
+    return false;
```

Comment:
ultra-nit: there's a triple negation at this point :) false to != arm64 to unnecessary. Can we simplify slightly?